### DISCUSSION TITLE
Model ERCOT region

### DISCUSSION THREAD

**Discussion by Edudiro** [pypsa-meets-earth/pypsa-earth#868] [Q&A] [answered]
Hi, I am not sure this is the right forum to ask non-coding questions, so please redirect me to the right place if that's the case, thank you.

I have been using quite a lot pypsa-eur and become quite comfortable with it. However, with the renewables expansion in the US, I have decided to try to model the ERCOT region, how can I choose such region in pypsa-earth? Selecting the ISO code for US and then reducing the region does not do the job because only the cutout and build_shape rules run me out of RAM memory :( 

Tried selecting the country as "US-TX", but it is not recognized.

Is there a way to model this, or am I running too fast?

Thank you for your time!

**Comment by ekatef** [comment, marked as answer]
Hello @Edudiro! That a definitely a great place for your questions as it directly relates to the model design. Sorry for taking a while to answer.

Currently, there is work ongoing to increase spatial resolution of the model for some regions of the whole modelled area: in particular, [this issue](https://github.com/pypsa-meets-earth/pypsa-earth/issues/788) considers a zooming into administrative region. Agree that it can lead to some issues linked with performance limitation. Not sure however if it would be worth to have an approach cutting down the considered region, as it would mean to neglect all the interconnections between the focus area and the rest of the power system.

As a workaround, I can suggest to supply custom shape files to describe a region of interest and commenting-out `build_shapes` rule. But it would require some extra-work, while I doubt if the modeling results would be adequate. If you task can be re-defined as a micro-grid problem, it may worth to have a look on [PyPSA-Distribution](https://github.com/pypsa-meets-earth/pypsa-distribution) which is less computationally expensive.

**Comment by davide-f** [comment]
For US, you may look at this draft PR: https://github.com/pypsa-meets-earth/pypsa-earth/pull/832
It contains some draft code that once finalized and merged would allow what you are interested into.
For more finer resolutions, PyPSA-Distribution may also be of your interest :)

**Comment by Edudiro** [comment]
Dear @davide-f and @ekatef, 
Thank you for your answer, from what I have understood, PR #832 dowloads the whole country and then crops the desired region. That would still require great memory allocation for some countries like the US. Would there be a way to handle only the specific regions?

Regardless, it is great progress and definetly what I was looking for, thank you.
